src/gleam_stdlib.erl

```diff
@@ -167,6 +167,16 @@ parse_float(String) ->
         _ -> {error, nil}
     end.
 
+random_int_range(Inclusive, Exclusive) ->
+    case {Inclusive, Exclusive} of
+        {_, _} when Inclusive =:= Exclusive ->
+            Inclusive;
+        {Max, Min} when Inclusive > Exclusive ->
+            Exclusive + rand:uniform(Max - Min);
+        {Min@1, Max@1} when Inclusive < Exclusive ->
+            Exclusive - rand:uniform(Max@1 - Min@1)
+    end.
```

Comment:
Isn't this the logic we just removed in the other PR?

Reply:
It's different, because `random() * {exclusive - inclusive} + inclusive` takes advantage of the fact thst `{exclusive - inclusive}` is allowd to be negative to remove cases.

rand:uniform only accepts positive integers, and so we must do these cases. However it gives an integer directly, so this still ends up being faster i believe.

Reply:
Makes sense! How does it benchmark against the non-external version?

Reply:
Some basic benchmarks for `random(3, 10)`. I'll also edit the main PR

```
performance_test_gleam_erlang:random_int_external/2   10000000     2.66   1242316  [      0.12]
performance_test_gleam_erlang:random_int_fallback/2   10000000     7.35   3432243  [      0.34]
```

```gleam
pub fn random_int_fallback(inclusive: Int, exclusive: Int) -> Int {
  let result = random_float(int.to_float(inclusive), int.to_float(exclusive))
  case inclusive < exclusive {
    True -> float.floor(result)
    False -> float.ceiling(result)
  }
  |> float.round()
}

pub fn random_int_external(inclusive: Int, exclusive: Int) -> Int {
  case inclusive, exclusive {
    _, _ if inclusive == exclusive -> inclusive
    max, min if inclusive > exclusive ->
      exclusive + do_random_uniform_int(max - min)
    min, max if inclusive < exclusive ->
      exclusive - do_random_uniform_int(max - min)
  }
}

pub fn random_float(inclusive: Float, exclusive: Float) -> Float {
  do_random_uniform() *. { exclusive -. inclusive } +. inclusive
}

@external(erlang, "rand", "uniform")
fn do_random_uniform() -> Float

@external(erlang, "rand", "uniform")
fn do_random_uniform_int(range: Int) -> Int
```